### ISSUE TITLE
How many epoch

### ISSUE THREAD

**Issue by minmax100** [JiahuiYu/generative_inpainting#53]
Could I know how many epoch you have used to train your model (e.g., release_places2_256 model)? Any suggest min epoch to re-produce your impainting result?

Also, I observe the loss kept increasing during my training (e.g., from -0.47 to -0.21 and -0.06 ...). Does that indicate that I am not on the right track? (I test the resulted model and they do have bad accuracy.) What parameter should I check if it is not normal? Thanks!

|####################| 100.00%, 74660/0 sec. train epoch 1, iter 10000/10000, loss -0.472547, 0.13 batches/sec.
|####################| 100.00%, 74743/0 sec. train epoch 2, iter 10000/10000, loss -0.217382, 0.13 batches/sec.
|####################| 100.00%, 74793/0 sec. train epoch 3, iter 10000/10000, loss -0.061427, 0.13 batches/sec.
|#################---| 87.40%, 65406/9362 sec. train epoch 4, iter 8740/10000, loss -0.031020, 0.13 batches/sec.

**Comment by JiahuiYu** [comment]
The loss does not matter. It is possible from -0.47 to -0.21 and -0.06.
I trained the network for about 120 hours, the speed on my machine with one GPU is 0.3 ~ 0.4 batches/sec.
I saw your logs saying 0.13 batches/sec, thus recommend train around two weeks.

For the resulted model, specifically how bad it is? If the color is totally incorrect or it generates random contents, it is possible some bugs exist. You can view tensorboard/images to see the progress of inpainting model for debugging.

**Comment by doublejtoh** [comment]
My custom training data consists of 1000 images. I trained these 32000 epochs, about 15 hours on Titan V gpu(0.6 batches/sec). Results seem to be quite bad.. in your opinion, Is this overfitted? Or should it be trained with more epochs?

**Comment by JiahuiYu** [comment]
Is the training inpainting results good or bad?

**Comment by doublejtoh** [comment]
<img width="938" alt="2018-05-31 9 06 56" src="https://user-images.githubusercontent.com/26649742/40781381-9ba3d3aa-6516-11e8-92e9-08cac4fdbcba.png">
<img width="820" alt="2018-05-31 9 06 48" src="https://user-images.githubusercontent.com/26649742/40781385-a1c9b3da-6516-11e8-97ca-c945fa025e62.png">
<img width="203" alt="2018-05-31 9 06 46" src="https://user-images.githubusercontent.com/26649742/40781386-a364358a-6516-11e8-9696-7a95e4bec929.png">
<img width="946" alt="2018-05-31 9 06 26" src="https://user-images.githubusercontent.com/26649742/40781397-aa641710-6516-11e8-8abb-8d1703ec78fe.png">
<img width="942" alt="2018-05-31 9 06 15" src="https://user-images.githubusercontent.com/26649742/40781400-ac45db68-6516-11e8-8588-412a60fb23ae.png">
<img width="916" alt="2018-05-31 9 06 09" src="https://user-images.githubusercontent.com/26649742/40781404-b062ec90-6516-11e8-8d64-f189e838ef6d.png">
<img width="925" alt="2018-05-31 9 06 00" src="https://user-images.githubusercontent.com/26649742/40781405-b0907b06-6516-11e8-8da4-330b27930974.png">

these are the results.. some imgs are good and some imgs are not good..

**Comment by doublejtoh** [comment]
Also on test data set, i found that small missing regions are well inpainted while large missing region inpainted results are too blurry.

**Comment by JiahuiYu** [comment]
It seems the GAN loss is too large. You can set higher pixel-wise loss by setting these hyper-parameters, like set to 2.0. Hyper-parameters [here](https://github.com/JiahuiYu/generative_inpainting/blob/a2637d141727b2a7f9c3d2087e92691538eceb85/inpaint.yml#L12-L14).

**Comment by doublejtoh** [comment]
#21 i referenced to your suggestion that if result blurry, set above hyper parameters to 0.5. I’ll retrain it with 2.0. THANKS.

**Comment by JiahuiYu** [comment]
Good finding. Thanks.

**Comment by DAA233** [comment]
Hi, @JiahuiYu,

I assume that the two-stage model should be run twice:
1. Run with hyper-parameter `PRETRAIN_COARSE_NETWORK: False` and get a pre-trained coarse model;
2. Run again with hyper-parameter `PRETRAIN_COARSE_NETWORK: True` and with `MODEL_RESTORE` set to the model path we get in the first stage.

Since you have mentioned "trained the network for about 120 hours", I wonder how many hours are used to train the coarse model. Thanks!

**Comment by JiahuiYu** [comment]
I train it jointly. Sorry for confusionabout the code.

**Comment by DAA233** [comment]
Thanks for the reply.

It means that if we set hyper-parameter `PRETRAIN_COARSE_NETWORK: True` then a two-stage model will be built and it will be trained jointly, right? I will try to get more familiar with the code.

**Comment by JiahuiYu** [comment]
@DAA233 Yes.

**Comment by guvcolie** [comment]
Thank you for your work!
You said "I train it jointly. Sorry for confusion about the code", do you mean that "PRETRAIN_COARSE_NETWORK should be always False"?

**Comment by JiahuiYu** [comment]
Yes. Since empirically I cannot distinguish the difference of performance when train jointly or separately. So currently PRETRAIN_COARSE_NETWORK is a legacy interface.

**Comment by guvcolie** [comment]
@JiahuiYu Got it! Thank you!
